### PR TITLE
Make -p honor test stanzas

### PR DESCRIPTION
**Pull request by emillon**
Previously, this option would consider that test stanzas do not have any package set (see #1231, cc @samoht).

### CODE REVIEW

File: src/gen_rules.ml [hunk 1]
```diff
@@ -359,6 +359,23 @@ module type Gen = sig
...
+      Some package
+  | _ -> None
+
+let relevant_stanzas pkgs stanzas =
+  let is_relevant stanza =
+    match stanza_package stanza with
+    | Some package -> Package.Name.Set.mem pkgs package.name
+    | None -> true
+  in
+  List.filter stanzas ~f:is_relevant
```
Comment (rgrinberg) [inline src/gen_rules.ml]: Personally, the intermediate `is_relevant` name doesn't really increase readability here.

**Review by rgrinberg — APPROVED** [review]
Looks good.

**Comment by emillon** [comment]
The extensible variant makes it difficult to update these pattern matches. Is this so that extensions can define their stanzas? If that's the case, how about we group the builtin stanzas under a closed variant type so that pattern matching is checked?

```
type builtin_stanza =
  | Executables 
  | ...
  | Tests

type Stanza.t = ...

type Stanza.t +=
  | Builtin of builtin_stanza
```

File: src/gen_rules.ml (hunk 1 above)
Comment (emillon) [reply src/gen_rules.ml]: right, it was on toplevel for a moment. I'll inline it back.

**Comment by rgrinberg** [comment]
I think that would be good. The other alternative would be to add a registration API to stanzas. In which case a:

```
val package : t -> Package.t option
```

Could be provided. This would solve the issue for plugin stanzas as well. It's not necessary until we add more plugins however.

**Pull request merged**